parsec/parsec_hwloc.c

```diff
@@ -78,43 +89,94 @@ static void parsec_hwloc_print_cpuset(int verb, char* msg, hwloc_cpuset_t cpuset
 int parsec_hwloc_init(void)
 {
 #if defined(PARSEC_HAVE_HWLOC)
-    if ( first_init ) {
+    if ( !parsec_hwloc_first_init ) {
```

Comment:
This not being thread-safe/mutexed crashes the class/atomics ctest (only test were we exercise multithreaded initialization of hwloc with a straight call to bind threads without prior initialization in parsec_init)

Reply:
Discussion about why the simpler resolution is correct (extracted from Slack)
> Aurelien Bouteiller [17:31] we have this global `parsec_hwloc_first_init` but we do all sort of non atomic updates to it so it is useless
[17:34](https://icldisco.slack.com/archives/GC2L4J95G/p1709332445810949)
I can fix the class/atomics test (just call parsec_hwloc_init from the main thread before doing pthread_create) but I don’t see how this variable is supposed to help, if we have threads its broken, if we don’t its useless?
> 
>George Bosilca
  [11:16](https://icldisco.slack.com/archives/GC2L4J95G/p1709482567324649)
I think you misunderstood the goal of `parsec_hwloc_first_init`. In a normal situation, before threads can do anything related to HWLOC, parsec_init has been called, and the entire HWLOC subsystem has been initialized (so the parsec_hwloc_first_initis > 0)
[11:17](https://icldisco.slack.com/archives/GC2L4J95G/p1709482632527469)
parsec_hwloc_first_init protects against multiple initialization of the HWLOC info in case we create multiple contexts in the same application. There was no need to use atomics, because from where this is reached there is no race condition possible, in a normal setup